sim/mage/arcane/arcane.go

```diff
@@ -34,16 +40,67 @@ func NewArcaneMage(character *core.Character, options *proto.Player) *ArcaneMage
 	return arcaneMage
 }
 
-type ArcaneMage struct {
-	*mage.Mage
-
-	Options *proto.ArcaneMage_Options
-}
-
 func (arcaneMage *ArcaneMage) GetMage() *mage.Mage {
 	return arcaneMage.Mage
 }
 
 func (arcaneMage *ArcaneMage) Reset(sim *core.Simulation) {
 	arcaneMage.Mage.Reset(sim)
 }
+
+func (arcaneMage *ArcaneMage) Initialize() {
+	arcaneMage.Mage.Initialize()
+
+	arcaneMage.registerArcaneBarrageSpell()
+}
+
+func (arcaneMage *ArcaneMage) ApplyTalents() {
+
+	arcaneMage.Mage.ApplyTalents()
+	// Arcane Specialization Bonus
+	arcaneMage.AddStaticMod(core.SpellModConfig{
+		School:     core.SpellSchoolArcane,
+		FloatValue: 0.25,
+		Kind:       core.SpellMod_DamageDone_Flat,
+	})
```

Comment:
This looks weird as a mod. Usually Additive mods target specific spells. I'm not sure how this specialization should work but if its supposed to work with non mage spells like trinkets/procs it should probably use `DamageDone_Pct`. If it is indeed supposed to only affect mage spells then Flat + a class mask for MageSpellsAll would be correct

Reply:
Looking at the spell in wowhead it does look like it hsould be `DamageDone_Pct` on all arcane spells not only mage ones